src/main/java/Main.java

```diff
@@ -2,19 +2,49 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Collectors;
 
 public class Main {
+
+    private static final Integer N_INDEXING_FILES = 100;
+
+    private static <T> List<List<T>> partition(List<T> collection, int partitionSize){
+        List<List<T>> partitions = new ArrayList<>();
+
+        for (int i = 0; i < collection.size(); i += partitionSize) {
+            partitions.add(collection.subList(i, Math.min(i + partitionSize, collection.size())));
+        }
+        return partitions;
+    }
+
     public static void main(String[] args) throws IOException {
+        ArticlesIndexer indexer = new ArticlesIndexer();
         Path folderPath = Paths.get("data");
         List<Path> pathStream = Files.walk(folderPath)
```

Comment:
I would review the use of a try-with-resources to ensury that the stream is properly close.